### PR TITLE
[rest] use correct Content-Type header for RESTful API

### PR DESCRIPTION
**Pull request by agners**
Set the content type to `text/plain` by default, and `application/json` where needed.

### CODE REVIEW

File: src/rest/response.hpp [hunk 1]
```diff
@@ -61,7 +64,22 @@ class Response
      *
      *
      */
-    Response(void);
+    Response();
```
Comment (jwhui) [inline src/rest/response.hpp:67]: nit: unintended change?

File: src/rest/response.hpp [hunk 2]
```diff
@@ -61,7 +64,22 @@ class Response
...
+
+    /**
+     * This method set the response Content-Type header.
```
Comment (jwhui) [inline src/rest/response.hpp:70]: nit:
```suggestion
     * This method sets the response Content-Type header.
```

File: src/rest/response.hpp [hunk 3]
```diff
@@ -61,7 +64,22 @@ class Response
...
+     *
+     * @param[in] aContentType  A string to be set as response Content-Type header.
+     *
+     */
+    void SetContentType(const std::string &aContentType);
+
+    /**
+     * This method return a string of the response Content-Type.
```
Comment (jwhui) [inline src/rest/response.hpp:78]: nit:
```suggestion
     * This method returns a string of the response Content-Type.
```

**Review by jwhui — COMMENTED** [review]
Thanks for contributing this! Commented on some minor nits.

**Review by bukepo — APPROVED** [review]

File: src/rest/response.cpp [hunk 4]
```diff
@@ -91,6 +85,16 @@ void Response::SetCallback(void)
...
 }
 
+void Response::SetContentType(const std::string &aContentType)
+{
+    mHeader[OT_REST_RESPONSE_CONTENT_TYPE_HEADER] = aContentType;
+}
+
+std::string Response::GetContentType() const
+{
+    return mHeader.find(OT_REST_RESPONSE_CONTENT_TYPE_HEADER)->second;
```
Comment (wgtdkp) [inline src/rest/response.cpp:95]: Why not access by `[]`?
```suggestion
    return mHeader[OT_REST_RESPONSE_CONTENT_TYPE_HEADER];
```

**Review by wgtdkp — APPROVED** [review]
Thanks! @agners 
LGTM! :+1:

**Comment by wgtdkp** [comment]
> Set the content type to `text/plain` by default, and `application/json` where needed.

Curious, are there cases where we need to use `text/plain`?

File: src/rest/response.cpp (hunk 4 above)
Comment (agners) [reply src/rest/response.cpp:95]: That does not work:

```
/home/sag/projects/openthread/ot-br-posix/src/rest/response.cpp: In member function ‘std::string otbr::rest::Response::GetContentType() const’:
/home/sag/projects/openthread/ot-br-posix/src/rest/response.cpp:93:34: error: passing ‘const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >’ as ‘this’ argument discards qualifiers [-fpermissive]
   93 |     return mHeader["Content-Type"];
      |                                  ^
```

Comment (agners) [reply src/rest/response.cpp:95]: More detailed explanation: https://stackoverflow.com/a/262872/184207

Maybe `at()` is the better option here as we know it must exist?

**Comment by agners** [comment]
> > Set the content type to `text/plain` by default, and `application/json` where needed.
> 
> Curious, are there cases where we need to use `text/plain`?

All `/node/*` (e.g. `/node/rloc` etc) endpoints return a single value only, not JSON encoded. They represent the majority of endpoints.

**Comment by wgtdkp** [comment]
> /node/rloc

@agners But a single integer is till a valid JSON value. right? https://www.json.org/json-en.html

**Comment by agners** [comment]
> > /node/rloc
> 
> @agners But a single integer is till a valid JSON value. right? https://www.json.org/json-en.html

Hm, right, single values are valid JSON. Strings need to be in quotes, and that seems to case.

I did not really had a practical reason for the content-type change, somehow in my mind just values was not valid JSON :sweat_smile:  

Where should we go from here? There is currently no need for an API to change content-type, so this makes the PR moot. I can separate the switch to the map data structure as it makes the code nicer and submit a new PR for that.

**Comment by wgtdkp** [comment]
> Where should we go from here? There is currently no need for an API to change content-type, so this makes the PR moot. I can separate the switch to the map data structure as it makes the code nicer and submit a new PR for that.

Let's keep all the RESTful APIs returning the same content-type. A separate PR which changes only the data structure sounds good to me! thanks!

**Comment by agners** [comment]
Data structure update PR #1665.

**Pull request closed without merging**